apps/src/code-studio/pd/workshop_dashboard/components/workshop_management.jsx

```diff
@@ -25,16 +31,16 @@ export default class WorkshopManagement extends React.Component {
     onDelete: null
   };
 
-  componentWillMount() {
-    if (this.props.showSurveyUrl) {
-      this.permission = new Permission();
+  constructor(props) {
+    super(props);
```

Comment:
What's this for? Does the base class have any constuctor logic?

Reply:
Yes, this is required because the base `React.Component` expects `props` in the constructor. This is a common React pattern. See https://reactjs.org/docs/state-and-lifecycle.html#adding-local-state-to-a-class